xbmc/windowing/gbm/drm/DRMUtils.cpp

```diff
@@ -160,79 +156,162 @@ bool CDRMUtils::FindPreferredMode()
   return true;
 }
 
-bool CDRMUtils::FindPlanes()
+// finds a video plane with given format + modifier, and a gui plane with inited format + modifier for any available crtcs
+bool CDRMUtils::FindPlanes(uint32_t format, uint64_t modifier)
 {
-  for (size_t i = 0; i < m_crtcs.size(); i++)
+  // current config already satisfies
+  if (m_gui_plane != nullptr && m_video_plane != nullptr &&
+      m_video_plane->SupportsFormatAndModifier(format, modifier))
+    return true;
+
+  uint32_t guiformat = m_gui_plane->GetFormat();
+  // loop over current crtc which is capable of rendering on the connected encoder (port)
+  for (size_t crtc_offset = 0; crtc_offset < m_crtcs.size(); crtc_offset++)
   {
-    if (!(m_encoder->GetPossibleCrtcs() & (1 << i)))
+    if (!(m_encoder->GetPossibleCrtcs() & (1 << crtc_offset)))
       continue;
+    // loop for each gui plane candidate which satisfies the current EGL rendered format
+    // gui format is decided during initialization, yet we can still swap planes if necessary
+    for (auto& gui_plane : m_planes)
+    {
+      auto gplane = gui_plane.get();
+      if (!(gplane->GetPossibleCrtcs() & (1 << crtc_offset)) ||
+          !gplane->SupportsFormatAndModifier(guiformat, DRM_FORMAT_MOD_LINEAR))
+        continue;
+      // loop for each format satisfying video plane candidate which is different than gui plane candidate
+      for (auto& vid_plane : m_planes)
+      {
+        auto vplane = vid_plane.get();
```

Comment:
The shorter names `gplane`, `vplane` and the `vid` prefix/suffix seem like an unnecessary abbreviation.